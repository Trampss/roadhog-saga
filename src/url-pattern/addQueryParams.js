import { encodeParams, isEmpty } from './utils'

export default url => (queryParams) => {
  if (isEmpty(queryParams)) return url

  const params = Object.keys(queryParams)
    .map((k => `${k}=${encodeParams(queryParams[k])}`))

  const slash = (params.length > 0 || url.endsWith('/') || url.endsWith('&') || url.endsWith('?')) ? '' : '/'
  const questionMark = (url.endsWith('&') || url.endsWith('?')) ? '' : '?'

  return `${url}${slash}${questionMark}${params.join('&')}`
}
