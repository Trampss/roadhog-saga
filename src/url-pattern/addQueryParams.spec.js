/* eslint-env jest */
import addQueryParams from './addQueryParams'

describe('addQueryParams', () => {
  it('should return url without modification', () => {
    expect(addQueryParams('http://an-url')()).toMatchSnapshot()
  })

  it('should return url with params (trailing /)', () => {
    expect(addQueryParams('http://an-url/')({ some: 'queries', to: 'add' })).toMatchSnapshot()
  })

  it('should return url with params (trailing &)', () => {
    expect(addQueryParams('http://an-url/?bla=yo&')({ some: 'queries', to: 'add' })).toMatchSnapshot()
  })

  it('should return url with params (trailing ?)', () => {
    expect(addQueryParams('http://an-url/?')({ some: 'queries', to: 'add' })).toMatchSnapshot()
  })

  it('should return url with params (no trailing)', () => {
    expect(addQueryParams('http://an-url')({ some: 'queries', to: 'add' })).toMatchSnapshot()
  })
})
