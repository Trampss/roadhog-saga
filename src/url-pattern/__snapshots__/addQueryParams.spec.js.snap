exports[`addQueryParams should return url with params (no trailing) 1`] = `"http://an-url/?some=queries&to=add"`;

exports[`addQueryParams should return url with params (trailing &) 1`] = `"http://an-url/?bla=yo&some=queries&to=add"`;

exports[`addQueryParams should return url with params (trailing /) 1`] = `"http://an-url/?some=queries&to=add"`;

exports[`addQueryParams should return url with params (trailing ?) 1`] = `"http://an-url/?some=queries&to=add"`;

exports[`addQueryParams should return url without modification 1`] = `"http://an-url"`;
